lib/s3.ts

```diff
@@ -0,0 +1,30 @@
+'use server'
+
+import { awsS3Bucket } from '@/config'
+import {
+  DeleteObjectCommand,
+  PutObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3'
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner'
+import { Image } from './schema'
+
+export async function createUploadUrl(key: string) {
+  return getSignedUrl(
+    new S3Client(),
+    new PutObjectCommand({
+      Bucket: awsS3Bucket,
+      Key: key,
+    }),
+  )
+}
+
+export async function deleteS3Image(image: Image) {
+  const key = new URL(image.src).pathname.slice(1)
+  return await new S3Client().send(
+    new DeleteObjectCommand({
+      Bucket: awsS3Bucket,
+      Key: key,
+    }),
+  )
+}
```

Comment:
Creating a new S3Client instance for each request can be inefficient. Consider reusing a single S3Client instance. Creating a new client for each request could potentially lead to resource exhaustion if many uploads are happening simultaneously.

```suggestion
const s3Client = new S3Client();

export async function createUploadUrl(key: string) {
  return getSignedUrl(
    s3Client,
    // rest of the function unchanged
  )
}

export async function deleteS3Image(image: Image) {
  // ...
  return await s3Client.send(
    // rest of the function unchanged
  )
}
```